### PR TITLE
Fix 100% crash when capturing instrumented functions.

### PR DESCRIPTION
**Pull request by pierricgimmig**
The code assumed that the hashmap contained the thread name. Use operator[] instead of at() to insert and return an empty string if the thread name is not (yet) present.

### CODE REVIEW

File: OrbitCore/CaptureData.h [hunk 1]
```diff
@@ -64,8 +64,8 @@ class CaptureData {
     return thread_names_;
   }
 
-  [[nodiscard]] const std::string& GetThreadName(int32_t thread_id) const {
-    return thread_names_.at(thread_id);
+  [[nodiscard]] const std::string& GetThreadName(int32_t thread_id) {
+    return thread_names_[thread_id];
```
Comment (dimitry-) [inline OrbitCore/CaptureData.h:68]: You can keep it const by doing find() -> return ""; in case it was not found. No need to modify the map.

Comment (pierricgimmig) [reply OrbitCore/CaptureData.h:68]: Then we'd have to change the return type. I guess using a static empty string would do the trick.

File: OrbitCore/CaptureData.h [hunk 2]
```diff
@@ -64,8 +64,10 @@ class CaptureData {
     return thread_names_;
   }
 
-  [[nodiscard]] const std::string& GetThreadName(int32_t thread_id) {
-    return thread_names_[thread_id];
+  [[nodiscard]] const std::string& GetThreadName(int32_t thread_id) const {
+    static std::string empty_string;
```
Comment (dimitry-) [inline OrbitCore/CaptureData.h:68]: kEmptyString, also const

File: OrbitCore/CaptureData.h [hunk 3]
```diff
@@ -64,8 +64,10 @@ class CaptureData {
...
+    return thread_names_.contains(thread_id) ? thread_names_.at(thread_id)
```
Comment (dimitry-) [inline OrbitCore/CaptureData.h:69]: doesn't this do 2 lookups instead of one if you use find?

File: OrbitCore/CaptureData.h [hunk 4]
```diff
@@ -65,7 +65,10 @@ class CaptureData {
   }
 
   [[nodiscard]] const std::string& GetThreadName(int32_t thread_id) const {
-    return thread_names_.at(thread_id);
+    static const std::string kEmptyString;
+    auto it = thread_names_.find(thread_id);
+    return it != thread_names_.end() ? thread_names_.at(thread_id)
```
Comment (dimitry-) [inline OrbitCore/CaptureData.h:70]: still 2 lookups. use it->second instead of at() call

**Pull request merged**